src/main/java/com/amadeus/Response.java

```diff
@@ -67,6 +66,9 @@ protected void parse(HTTPClient client) {
   // Detects of any exceptions have occured and throws the appropriate exceptions.
   protected void detectError(HTTPClient client) throws ResponseException {
     ResponseException exception = null;
+    if (statusCode == 204) {
```

Comment:
should it be part of the if else if else if part instead of being an if isolated?